components/http/okHttp/src/main/java/com/microsoft/kiota/http/OkHttpRequestAdapter.java

```diff
@@ -907,30 +906,19 @@ public MediaType contentType() {
 
                                 @Override
                                 public long contentLength() throws IOException {
-                                    long length;
                                     final Set<String> contentLength =
                                             requestInfo.headers.getOrDefault(
                                                     contentLengthHeaderKey, new HashSet<>());
-                                    if (contentLength.isEmpty()
-                                            && requestInfo.content
-                                                    instanceof ByteArrayInputStream) {
+                                    if (!contentLength.isEmpty()) {
+                                        return Long.parseLong(
+                                                contentLength.toArray(new String[] {})[0]);
+                                    }
+                                    if (requestInfo.content instanceof ByteArrayInputStream) {
                                         final ByteArrayInputStream contentStream =
                                                 (ByteArrayInputStream) requestInfo.content;
-                                        length = contentStream.available();
-                                    } else {
-                                        length =
-                                                Long.parseLong(
-                                                        contentLength.toArray(new String[] {})[0]);
-                                    }
-                                    if (length <= 0) {
-                                        length = super.contentLength();
-                                    }
-                                    if (length > 0) {
-                                        spanForAttributes.setAttribute(
-                                                HttpIncubatingAttributes.HTTP_REQUEST_BODY_SIZE,
-                                                length);
+                                        return contentStream.available();
                                     }
```

Comment:
From the "summarized" issue
> @andreaTP rightfully pointed out that [we should not be using the available method](https://stackoverflow.com/a/1119346/3808675) since implementations behaviour vary. I think that part of the confusion here is that streams in dotnet behave a bit differently w.r.t. that aspect.

I haven't tested the latest version of the code at this point. And it's likely that in most cases a byte array input stream which hasn't been read will return the total payload length when reading available. But to be safe, are we sure we absolutely need this block since super.contentLength has been restored?

Reply:
Tests fail because `super.contentLength()` sets [`-1`](https://github.com/square/okhttp/blob/master/okhttp/src/main/kotlin/okhttp3/internal/-RequestBodyCommon.kt#L43) by default on the Request object. 

Okhttp probably always adds the right content-length when no overrides are used but I'm unable to unit test for it, even at the interceptors, the Content-Length is not yet added as a header. Might you know how else I can test?

Reply:
thanks for testing this. Can you at least add a comment for context for this block please?
And no, I'm not aware of any other way of testing besides maybe setting up a mock server in the unit test and testing what the mock server receives. But even then, the telemetry attribute would always be wrong...